sebs/knative/knative.py

```diff
@@ -0,0 +1,552 @@
+        with open(yaml_out, "w") as yaml_file:
+            yaml.dump(func_yaml_content, yaml_file, default_flow_style=False)
+
+        # Create Procfile for Python runtime
+        if language_name == "python":
+            procfile_content = "web: python3 -m parliament ."
+            procfile_out = os.path.join(directory, "Procfile")
+            with open(procfile_out, "w") as procfile_file:
+                procfile_file.write(procfile_content)
+
+            # Create an empty __init__.py file
+            init_file_out = os.path.join(directory, "__init__.py")
+            open(init_file_out, "a").close()
+        
+        # Modify package.json for Node.js runtime to add the faas-js-runtime.
+        if language_name == "nodejs":
+            package_json_path = os.path.join(directory, "package.json")
+            if os.path.exists(package_json_path):
+                with open(package_json_path, "r+") as package_file:
+                    package_data = json.load(package_file)
+                    if "scripts" not in package_data:
+                        package_data["scripts"] = {}
+                    package_data["scripts"]["start"] = "faas-js-runtime ./index.js"
+                    package_file.seek(0)
+                    package_file.write(json.dumps(package_data, indent=2))
+                    package_file.truncate()
+
+        package_config = CONFIG_FILES[language_name]
+        function_dir = os.path.join(directory, "function")
+        os.makedirs(function_dir)
+        # move all files to 'function' except func.py, func.yaml, Procfile
+        for file in os.listdir(directory):
+            if file not in package_config:
+                file = os.path.join(directory, file)
+                shutil.move(file, function_dir)
+
+        self.build_base_image(
+            directory, language_name, language_version, benchmark, is_cached
+        )
+
+        code_size = Benchmark.directory_size(directory)
+        return directory, code_size
+
+    def storage_arguments(self) -> List[str]:
+        storage = cast(KnativeMinio, self.get_storage())
+        return [
+            "-p",
+            "MINIO_STORAGE_SECRET_KEY",
+            storage.config.secret_key,
+            "-p",
+            "MINIO_STORAGE_ACCESS_KEY",
+            storage.config.access_key,
+            "-p",
+            "MINIO_STORAGE_CONNECTION_URL",
+            storage.config.address,
+        ]
+
+    def create_function(
+        self, code_package: Benchmark, func_name: str
+    ) -> "KnativeFunction":
+        self.logging.info("Building Knative function")
+        try:
+            # Check if the function already exists
+            knative_func_command = subprocess.run(
+                [*self.get_knative_func_cmd(), "list"],
+                stderr=subprocess.DEVNULL,
+                stdout=subprocess.PIPE,
+            )
+            sanitize_benchmark_name = sanitize_benchmark_name_for_knative(
```

Comment:
This function should be a member of this class - what is the purpose of moving it to an unrelated module?